src/fs/block_allocator.hh

```diff
@@ -0,0 +1,39 @@
+/*
+ * This file is open source software, licensed to you under the terms
+ * of the Apache License, Version 2.0 (the "License").  See the NOTICE file
+ * distributed with this work for additional information regarding copyright
+ * ownership.  You may not use this file except in compliance with the License.
+ *
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+/*
+ * Copyright (C) 2019 ScyllaDB
+ */
+
+#pragma once
+
+#include <unordered_map>
+#include <queue>
+
+namespace seastar {
+
+class block_allocator {
+    std::unordered_map<size_t, bool> _block_pool;
+    std::queue<size_t> _free_blocks;
+public:
+    block_allocator(std::unordered_map<size_t, bool>, std::queue<size_t>);
```

Comment:
What is the purpose of passing that unordered_map to the block_allocator constructor?

Reply:
for testing, will change